### PR TITLE
Change to use global tool/test tool

### PR DESCRIPTION
**Pull request by andyhopp**
*Issue #, if available:*

*Description of changes:*
This PR includes changes to use the .NET Lambda global tool for packaging as well as the dotnet-lambda-testtool-3.1 for testing.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-dotnet-function-runtime-provider/src/utils/detect.ts [hunk 1]
```diff
@@ -15,21 +16,63 @@ export const detectDotNetCore = async (): Promise<CheckDependenciesResult> => {
...
+
+async function installGlobalTool(toolName: string) {
+  let response = await inquirer.prompt({
+    type: 'confirm',
+    name: 'installToolkit',
+    message: `The ${toolName} global tool is required but was not detected.\nWould you like to install this tool?`,
+    default: 'Y',
+  });
+  if (response.installToolkit) {
+    let toolInstallationResult = execa.sync(executableName, ['tool', 'install', '-g', 'Amazon.Lambda.Tools']);
```
Comment (edwardfoyle) [inline packages/amplify-dotnet-function-runtime-provider/src/utils/detect.ts:73]: should `'Amazon.Lambda.Tools'` be `toolName`?

File: packages/amplify-dotnet-function-runtime-provider/src/utils/detect.ts [hunk 2]
```diff
@@ -15,21 +16,63 @@ export const detectDotNetCore = async (): Promise<CheckDependenciesResult> => {
...
+  const installedToolList = toolResult.stdout;
+
+  if (toolResult.exitCode !== 0) {
+    throw new Error(`${executableName} failed tool detection, exit code was ${toolResult.exitCode}`);
+  }
+
+  let toolInstalled = false;
+  let testToolInstalled = false;
+  if (installedToolList) {
+    if (installedToolList.match(/^amazon\.lambda\.tools/m)) {
```
Comment (edwardfoyle) [inline packages/amplify-dotnet-function-runtime-provider/src/utils/detect.ts:37]: Just want to confirm that the regex is checking for `'amazon.lambda.tools'` but the installer is installing `'Amazon.Lambda.Tools'` (difference in caps)

File: packages/amplify-dotnet-function-runtime-provider/src/utils/detect.ts [hunk 3]
```diff
@@ -15,21 +16,63 @@ export const detectDotNetCore = async (): Promise<CheckDependenciesResult> => {
...
+
+  if (!toolInstalled) {
+    await installGlobalTool('Amazon.Lambda.Tools');
+  }
+  if (!testToolInstalled) {
+    await installGlobalTool('Amazon.Lambda.TestTool-3.1');
+  }
+
+  // Verify that a dotnet 3.1 SDK and the dotnet Lambda tools is installed locally
+  if (sdkInstalled && toolInstalled) {
```
Comment (edwardfoyle) [inline packages/amplify-dotnet-function-runtime-provider/src/utils/detect.ts:53]: should this condition also include testToolInstalled?

File: packages/amplify-dotnet-function-runtime-provider/src/utils/detect.ts (hunk 2 above)
Comment (andyhopp) [reply packages/amplify-dotnet-function-runtime-provider/src/utils/detect.ts:37]: The results from the CLI are all-lowercase.

File: packages/amplify-dotnet-function-runtime-provider/src/utils/detect.ts (hunk 3 above)
Comment (andyhopp) [reply packages/amplify-dotnet-function-runtime-provider/src/utils/detect.ts]: Yep! Fixed.

File: packages/amplify-dotnet-function-runtime-provider/src/utils/detect.ts (hunk 1 above)
Comment (andyhopp) [reply packages/amplify-dotnet-function-runtime-provider/src/utils/detect.ts]: Eek! Yes, that is a bug. Fixed!

**Pull request merged**